ArduCopter/Copter.cpp

```diff
@@ -656,12 +656,13 @@ Copter::Copter(void)
     land_accel_ef_filter(LAND_DETECTOR_ACCEL_LPF_CUTOFF),
     rc_throttle_control_in_filter(1.0f),
     inertial_nav(ahrs),
-    param_loader(var_info),
-    flightmode(&mode_stabilize)
+    param_loader(var_info)
 {
     // init sensor error logging flags
     sensor_health.baro = true;
     sensor_health.compass = true;
+
+    flightmode = mode_from_mode_num(control_mode);
```

Comment:
This should be the other way around, setting control_mode from flight_mode.

Reply:
@peterbarker san.
Thanks for your comment.
I will add the mode_from_mode_num method mode_num_from_mode method.
I will try to get the mode number from the mode pointer.

Reply:
I changed the flight mode pointer to set the control mode.